src/test/java/com/epam/eighty/repository/CustomerRepositoryTest.java

```diff
@@ -30,121 +28,71 @@
+        engine.execute(creatCypherScript);
     }
 
+    @After
+    public void cleanTestDatabase() {
+        engine.execute(deleteScript);
+    }
+    
     @Test
-    public void test_getSortedSetOfCustomersByName() {
-        Customer c1 = new Customer();
-        c1.setName("fake customer 1");
-        customerRepo.save(c1);
-        Customer c2 = new Customer();
-        c2.setName("fake customer 2");
-        customerRepo.save(c2);
-
-        Slice<Customer> customers = customerRepo.getSortedSetOfCustomersByName("fake customer.*");
-        assertNotNull(customers);
-        assertEquals(customers.getContent().size(), 2);
-        customerRepo.delete(c1);
-        customerRepo.delete(c2);
+    public void test_getCustomerByTopicId() throws IOException {
+        Long rootTopicId = topicRepo.findBySchemaPropertyValue("title", "root").get().getId();
+        List<Customer> customers = customerRepo.getCustomersByTopicId(rootTopicId);
+        assertEquals(4, customers.size());
+        
+        Long subTopicId = topicRepo.findBySchemaPropertyValue("title", "Annotations").get().getId();
+        customers = customerRepo.getCustomersByTopicId(subTopicId);
+        assertEquals(0, customers.size());
+        
+        customers = customerRepo.getCustomersByTopicId(9999L);
+        assertEquals(0, customers.size());
     }
 
     @Test
-    public void test_getQuestionsInTopicByCustomer() {
-        Topic root = new Topic();
-        root.setTitle("root");
-
-        Question q1 = new Question();
-        q1.setQuestion("q1");
-        Question q2 = new Question();
-        q2.setQuestion("q2");
-
-        Customer customer = new Customer();
-        customer.setName("customer");
-
-        Set<Customer> customers = new HashSet<>();
-        customers.add(customer);
-
-        q1.setCustomers(customers);
-        q2.setCustomers(customers);
-
-        Set<Question> questions = new HashSet<>();
-        questions.add(q1);
-        questions.add(q2);
-
-        root.setQuestions(questions);
-        topicRepo.save(root);
-
-        long questionCount = customerRepo.getQuestionsInTopicByCustomer(customer.getName(), root.getId());
-        assertEquals(questionCount, 2);
-
-        customerRepo.delete(customer);
-        questionRepo.delete(q1);
-        questionRepo.delete(q2);
-        topicRepo.delete(root);
+    public void test_getSortedSliceOfCustomersByName() {
+        List<Customer> customers = customerRepo.getSortedCustomersMatchingName("stom");
+        assertEquals(4, customers.size());
+        
+        assertEquals("Customer1", customers.get(0).getName());
+        assertEquals("Customer2", customers.get(1).getName());
+        assertEquals("Customer3", customers.get(2).getName());
+        assertEquals("Customer4", customers.get(3).getName());
+        
+        customers = customerRepo.getSortedCustomersMatchingName("fake");
+        assertEquals(0, customers.size());
     }
 
     @Test
-    public void test_getAllCustomers() {
```

Comment:
We didn't miss code coverage because we test so-called third-part implementation of our repository interfaces and cobertura doesn't include them in report